tests/parser-test-files/parse-expression.txt

```diff
@@ -0,0 +1,299 @@
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests)
+
+(define f 5)
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests)
+
+(define (f _x) 5)
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests)
+
+(define (f _x _y) 5)
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define f
+  (fn (x)
+    (+ x x)))
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define f
+  (fn ()
+    (+ 1 2)))
+
+================================================================================
+Parse expression (:disable)
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define (f)
+  (let ((x 1)
+        (declare y (T -> T))
+        (y (+ 1 2)))
+    (f x y)))
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define (f a b c)
+  (lisp (Integer -> Integer) (a b c)
+    (cl:+ 1 2)))
+
+================================================================================
+Parse expression (:disable)
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define (f x)
+  (match x
+    ((Some (Some x)) x)
+    ((Some x) x)
+    (y 1)
+    (1 z)
+    (_ 2)))
+
+================================================================================
+Parse expression (:disable)
+================================================================================
+
+(package coalton-unit-tests)
+
+(define (g x)
+  (match x))
+
+================================================================================
+Parse expression (:disable)
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define (f)
+  (let x = 1)
+  (let (Some y) = 2)
```

Comment:
```
(define (f)
    (let x = 1)
    (let ((y (from-some "invalid y" (Some 2))))
      (+ x y)))
```

This will test both let forms- I don't believe you can `let` bind a constructor like this